nuitka/tools/release/Documentation.py

```diff
@@ -28,68 +28,65 @@
 from nuitka.utils.FileOperations import getFileList
 
 
+def which(prog):
+    """
+    cross version program detection.
+
+    Whenever Nuitka decides to drop Python versions older than 3.3
+    the code in the else block can be removed.
+    """
+    def real(f):
+        def wrap(*args):
+            cmd = ""
+            if sys.version_info > (3, 3):
+                import shutil
+                cmd = shutil.which(prog)
+            else:
+                _, cmdname = os.path.split(prog)
+
+                for path in os.environ["PATH"].split(os.pathsep):
+                    cmd_ = os.path.join(path, cmdname)
+                    if os.path.isfile(cmd_) and os.access(cmd_, os.X_OK):
+                        cmd = cmd_
+            if cmd:
+                f(*args)
+            else:
+                print("Could not find such program %s" % prog)
+                sys.exit(1)
+        return wrap
+    return real
+
+
+@which('optipng')
+def optimize_pngs(pngList):
+    cmd = "optipng -o2 %s.png"
```

Comment:
Why not use this form of calling:

```
    subprocess.check_call(
        [
            "git",
            "update-index",
            "--cacheinfo",
            "%s,%s,%s"
            % (diff_entry["dst_mode"], new_object_hash, diff_entry["src_path"]),
        ]
    )
```

The shell need not be used to locate the program, the first argument is searched in `PATH` automatically I think. We only use the function to locate binaries, only for other purposes, not to execute it. Also, this means your quoting is perfect, because it's not needed. The shell will do funny things with spaces, etc. never a good idea to use it, unless you are lazy and want e.g. shell patterns.

Reply:
I can change this to the list format if you prefer.